source/acquaman/AMScanConfiguration.cpp

```diff
@@ -137,3 +141,28 @@ void AMScanConfiguration::setAxisControlInfos(const AMControlInfoList &axisContr
 		emit axisControlInfosChanged();
 	}
 }
+
+void AMScanConfiguration::setAutomaticDirectionAssessment(bool isAutomatic)
+{
+	if (automaticDirectionAssessment_ != isAutomatic){
+
+		automaticDirectionAssessment_ = isAutomatic;
+		emit automaticDirectionAssessmentChanged(automaticDirectionAssessment_);
+		setModified(true);
+	}
+}
+
+void AMScanConfiguration::setDirection(AMScanConfiguration::Direction newDirection)
+{
+	if (direction_ != newDirection){
+
+		direction_ = newDirection;
+		emit directionChanged(direction_);
+		setModified(true);
+	}
+}
+
+void AMScanConfiguration::setDirection(int newDirection)
+{
+	setDirection(Direction(newDirection));
```

Comment:
Is there a danger here of having a direction set that isn't one of the Direction options? I'm not sure what would happen with the cast in this case.